modules/common/logging/client.nix

```
@@ -6,29 +6,69 @@
   ...
 }:
 let
+  inherit (lib)
+    mkIf
+    mkEnableOption
+    mkOption
+    types
+    optionalString
+    ;
   cfg = config.ghaf.logging.client;
   inherit (config.ghaf.logging) listener;
 in
 {
   options.ghaf.logging.client = {
-    enable = lib.mkEnableOption "Enable the alloy client service";
-    endpoint = lib.mkOption {
+    enable = mkEnableOption "Enable the alloy client service";
+    endpoint = mkOption {
       description = ''
         Assign endpoint url value to the alloy.service running in
         different log producers. This endpoint URL will include
         protocol, upstream, address along with port value.
       '';
-      type = lib.types.str;
-      default = "http://${listener.address}:${toString listener.port}/loki/api/v1/push";
+      type = types.str;
+      default = "https://${listener.address}:${toString listener.port}/loki/api/v1/push";
+    };
+
+    tls = {
+      caFile = mkOption {
+        type = types.nullOr types.str;
```

As `caFile` is using path to file, so it will be good to `path` instead of `str`
```suggestion
        type = types.nullOr types.path;
```
Same goes to `certFile` and `KeyFile`

---

Done. I changed for both `client` and `server`.